File: algoritmos/2/src/model/Animal.ts
export abstract class Animal {
    private peso: number;
    private qtdPatas: number;

    constructor(peso: number, qtdPatas: number) {
        this.peso = peso;
        this.qtdPatas = qtdPatas;
    }


    andar() {
        console.log(`${this.constructor.name} andando...`);
    }

    comer() {
        console.log(`${this.constructor.name} comendo...`);
    }
}

File: algoritmos/TesteTecnico.Raltec02/TesteTecnico.Raltec02.Domain/Models/Base/Animal.cs
namespace TesteTecnico.Raltec02.Domain.Models.Base
{
	public abstract class Animal
	{
		public double Peso { get; set; }
		public int QuantidadePatas { get; set; }

		public virtual void Andar()
		{
			Console.WriteLine("estou andando.");
		}

		public virtual void Comer()
		{
			Console.WriteLine("estou comendo.");
		}
	}
}
